rosetta-lang/src/main/java/com/regnosys/rosetta/validation/RosettaSimpleValidator.xtend

```diff
@@ -1499,15 +1503,15 @@ class RosettaSimpleValidator extends AbstractDeclarativeRosettaValidator {
 	}
 	
 	private def void checkInputType(RosettaUnaryOperation o, RType type) {
-		if (!o.argument.getRType.isSubtypeOf(type)) {
+		if (!o.argument.getRMetaAnnotatedType.isSubtypeOf(type)) {
 			error('''Input type must be a «type».''', o, ROSETTA_OPERATION__OPERATOR)
 		}
 	}
 
 	private def void checkInputIsComparable(RosettaUnaryOperation o) {
-		val inputRType = o.argument.getRType
-		if (!inputRType.hasNaturalOrder) {
-			error('''Operation «o.operator» only supports comparable types (string, int, string, date). Found type «inputRType.name».''', o, ROSETTA_OPERATION__OPERATOR)
+		val inputRType = o.argument.getRMetaAnnotatedType
+		if (!inputRType.RType.hasNaturalOrder) {
+			error('''Operation «o.operator» only supports comparable types (string, int, string, date). Found type «inputRType.RType.name».''', o, ROSETTA_OPERATION__OPERATOR)
```

Comment:
Just saw this typo: should be `(string, int, number, boolean, date)`

Reply:
fixed